cordapp-example/workflows-kotlin/src/main/kotlin/com/example/utils/MoneyFinder.kt

```diff
@@ -9,35 +9,36 @@ import net.corda.core.node.services.Vault
 import net.corda.core.node.services.vault.PageSpecification
 import net.corda.core.node.services.vault.QueryCriteria
 import net.corda.core.node.services.vault.builder
+import java.lang.Exception
 
 // just for this exercise and make it simple `ExampleFlow` creates a money, so on payback I can use
 // "the same" money (same owner, same value). This class finds it.
-object MoneyFinder {
-    fun call(serviceHub: ServiceHub, cashOwner: Party, amount: Int): StateAndRef<CashState> {
-        val pageSpec = PageSpecification(pageNumber = 0, pageSize = 1)
-        val criteria = moneyQueryCriteria(cashOwner, amount.toLong())
-        val results = serviceHub.vaultService.queryBy(CashState::class.java, criteria = criteria, paging = pageSpec)
+fun moneyFinder(serviceHub: ServiceHub, cashOwner: Party, amount: Int): StateAndRef<CashState>? {
```

Comment:
Return a list instead because to reach the amount, you may have to add instances.